salt/cloud/clouds/clc.py

```diff
@@ -0,0 +1,368 @@
+    to refer to the salt-cloud driver that provides the underlying functionality to
+    connect to a cloud provider, while cloud profile configuration continues to use
+    ``provider`` to refer to the cloud provider configuration that you define.
+
+'''
+
+# Import python libs
+from __future__ import absolute_import
+from random import randint
+from re import findall
+import pprint
+import logging
+import time
+import os.path
+import subprocess
+import json
+# Import salt libs
+import salt.utils
+import salt.utils.cloud
+import salt.utils.xmlutil
+#import pip
+import importlib
+from salt.exceptions import SaltCloudSystemExit
+from flask import Flask, request
+# Import salt cloud libs
+import salt.config as config
+import requests
+# Attempt to import clc-sdk lib
+try:
+        importlib.import_module('clc')
+        import clc
+	HAS_CLC = True
+except ImportError:
+        HAS_CLC = False
+# Disable InsecureRequestWarning generated on python > 2.6
+try:
+    from requests.packages.urllib3 import disable_warnings  # pylint: disable=no-name-in-module
+    disable_warnings()
+except Exception:
+    pass
+
+try:
+    import salt.ext.six as six
+    HAS_SIX = True
+except ImportError:
+    # Salt version <= 2014.7.0
+    try:
+        import six
+    except ImportError:
+        HAS_SIX = False
+
+IP_RE = r'^(?:(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.){3}(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)$'
+
+# Get logging started
+log = logging.getLogger(__name__)
+
+__virtualname__ = 'clc'
+
+
+# Only load in this module if the CLC configurations are in place
+def __virtual__():
+    '''
+    Check for CLC configuration and if required libs are available.
+    '''
+    if get_configured_provider() is False:
+       return False
+
+    if get_dependencies() is False:
+        return False
+
+    return __virtualname__
+
+
+def get_configured_provider():
+    '''
+    Return the first configured instance.
+    '''
+    return config.is_provider_configured(
+        __opts__,
+        __active_provider_name__ or __virtualname__,
+        ('token','token_pass', 'user', 'password',)
+    )
+
+
+def get_dependencies():
+    '''
+    Warn if dependencies aren't met.
+    '''
+    deps = {
+        'clc': HAS_CLC,
+        'six': HAS_SIX
+    }
+    return config.check_driver_dependencies(
+        __virtualname__,
+        deps
+    )
+
+def get_creds():
+    user = config.get_cloud_config_value(
+    'user', get_configured_provider(), __opts__, search_global=False
+    )
+    password = config.get_cloud_config_value(
+    'password', get_configured_provider(), __opts__, search_global=False
+    )
+    token = config.get_cloud_config_value(
+    'token', get_configured_provider(), __opts__, search_global=False
+    )
+    token_pass = config.get_cloud_config_value(
+    'token_pass', get_configured_provider(), __opts__, search_global=False
+    )
+    creds = {"user":user,"password":password,"token":token,"token_pass":token_pass}
```

Comment:
Please add spaces here after the commas and use single-quotes, per our style guide.